src/mlpack/bindings/julia/get_julia_type.hpp

```diff
@@ -0,0 +1,136 @@
+/**
+ * @file get_julia_type.hpp
+ * @author Ryan Curtin
+ *
+ * Get the Julia-named type of an mlpack C++ type.
+ */
+#ifndef MLPACK_BINDINGS_JULIA_GET_JULIA_TYPE_HPP
+#define MLPACK_BINDINGS_JULIA_GET_JULIA_TYPE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace bindings {
+namespace julia {
+
+template<typename T>
+inline std::string GetJuliaType(
+    const typename std::enable_if<!util::IsStdVector<T>::value>::type* = 0,
+    const typename std::enable_if<!arma::is_arma_type<T>::value>::type* = 0,
+    const typename std::enable_if<!std::is_same<T,
+        std::tuple<data::DatasetInfo, arma::mat>>::value>::type* = 0,
+    const typename std::enable_if<!data::HasSerialize<T>::value>::type* = 0)
+{
+  return "unknown_"; // This will cause an error most likely...
+}
```

Comment:
This function gets called for model types which causes this:
```
`input_model::unknown_`: Input AdaBoost model.
```
I fixed the same bug in java bindings by adding
```
const typename std::enable_if<!std::is_same<T,
        std::tuple<data::DatasetInfo, arma::mat>>::value>::type* = 0
```
to the function that is meant to be called with serializable types (the last one in his file)

Reply:
This overload should only be called for classes that:

 * aren't one of the specific types in the template specifications below: `bool`, `int`, `size_t`, `double`, `std::string`
 * aren't `std::vector<T>`
 * aren't Armadillo types
 * aren't `std::tuple<data::DatasetInfo, arma::mat>`
 * don't have a `serialize()` method

So, actually, this should never be called.  As far as I can reason, adding `const typename std::enable_if<!std::is_same<T, std::tuple<data::DatasetInfo, arma::mat>>::value>::type*` shouldn't be necessary for the overload that gets models anyway, since `std::tuple<data::DatasetInfo, arma::mat>` won't have a `serialize()` member.

At least, I've certainly never seen any issues with this code in the generated Julia bindings.  Maybe the Java bindings are different---I haven't looked at them yet. :+1: